### PR TITLE
Add Esc-Esc to clear prompt when it's not empty

### PR DESCRIPTION
**Pull request by Adib234**
## Summary

As title suggests
## Details
Decision was made during UX meeting

## How to Validate
If you write text in the prompt and press `Esc-Esc` it should clear it. If the prompt is empty and you run `Esc-Esc` then it should run the slash command `/rewind`

## Pre-Merge Checklist

- [ ] Updated relevant documentation and README (if needed)
- [ ] Added/updated tests (if needed)
- [ ] Noted breaking changes (if any)
- [ ] Validated on required platforms/methods:
  - [ ] MacOS
    - [ ] npm run
    - [ ] npx
    - [ ] Docker
    - [ ] Podman
    - [ ] Seatbelt
  - [ ] Windows
    - [ ] npm run
    - [ ] npx
    - [ ] Docker
  - [ ] Linux
    - [ ] npm run
    - [ ] npx
    - [ ] Docker

### CODE REVIEW

File: packages/cli/src/ui/components/InputPrompt.tsx [hunk 1]
```diff
@@ -506,9 +506,14 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
...
-          // Second ESC triggers rewind
+          // Second ESC
           resetEscapeState();
-          onSubmit('/rewind');
+          if (keyMatchers[Command.CLEAR_INPUT](key) && buffer.text.length > 0) {
+            buffer.setText('');
+            resetCompletionState();
+          } else {
+            onSubmit('/rewind');
+          }
```
[the remark this answers is not in the document]
Comment (Adib234) [reply packages/cli/src/ui/components/InputPrompt.tsx]: Updated to remove trim() and clear whitespaces if they exist on esc-esc

File: docs/cli/keyboard-shortcuts.md [hunk 2]
```diff
@@ -34,7 +34,7 @@ available combinations.
 | ------------------------------------------------ | --------------------------------------------------------- |
 | Delete from the cursor to the end of the line.   | `Ctrl + K`                                                |
 | Delete from the cursor to the start of the line. | `Ctrl + U`                                                |
-| Clear all text in the input field.               | `Ctrl + C`                                                |
+| Clear all text in the input field.               | `Esc`                                                     |
```
[the remark this answers is not in the document]
Comment (Adib234) [reply docs/cli/keyboard-shortcuts.md]: Restored Ctrl-C

**Comment by Adib234** [comment]
> do we want both `Ctrl+C` and `Esc-esc` to clear input?

yep!

File: packages/cli/src/ui/components/InputPrompt.tsx [hunk 3]
```diff
@@ -506,9 +506,14 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
             resetEscapeState();
           }, 500);
         } else {
-          // Second ESC triggers rewind
+          // Second ESC
           resetEscapeState();
-          onSubmit('/rewind');
+          if (keyMatchers[Command.ESCAPE](key) && buffer.text.length > 0) {
```
Comment (jacob314) [inline packages/cli/src/ui/components/InputPrompt.tsx]: we do we check for the escape key press for the if case but not the else case? Would think both should be the same except for the buffer.text.length check

File: packages/cli/src/ui/components/StatusDisplay.tsx [hunk 4]
```diff
@@ -45,7 +45,12 @@ export const StatusDisplay: React.FC<StatusDisplayProps> = ({
   }
 
   if (uiState.showEscapePrompt) {
-    return <Text color={theme.text.secondary}>Press Esc again to rewind.</Text>;
+    const isPromptEmpty = uiState.buffer.text.length === 0;
+    return (
+      <Text color={theme.text.secondary}>
+        Press Esc again to {isPromptEmpty ? 'rewind' : 'clear prompt'}.
```
Comment (jacob314) [inline packages/cli/src/ui/components/StatusDisplay.tsx:57]: We should not show either message if there is no history to rewind yet.

File: packages/cli/src/ui/components/InputPrompt.tsx [hunk 5]
```diff
@@ -506,9 +506,14 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
...
+            buffer.setText('');
+            resetCompletionState();
+          } else {
+            onSubmit('/rewind');
```
Comment (jacob314) [inline packages/cli/src/ui/components/InputPrompt.tsx]: we should avoid triggering rewind unless there is actually somewhere to rewind to just as we avoid triggering clear unless there is something to clear.

File: packages/cli/src/ui/components/StatusDisplay.test.tsx [hunk 6]
```diff
@@ -147,7 +148,7 @@ describe('StatusDisplay', () => {
     expect(lastFrame()).toMatchSnapshot();
   });
 
-  it('renders Escape prompt', () => {
```
Comment (jacob314) [inline packages/cli/src/ui/components/StatusDisplay.test.tsx:150]: should also test the prompt rendered when the buffer is empty.

File: packages/cli/src/ui/components/InputPrompt.test.tsx [hunk 7]
```diff
@@ -1911,6 +1912,26 @@ describe('InputPrompt', () => {
       unmount();
     });
 
+    it('should clear the buff on esc esc if it has text', async () => {
```
Comment (jacob314) [inline packages/cli/src/ui/components/InputPrompt.test.tsx]: buff->buffer

**Review by jacob314 — COMMENTED** [review]
Approved after these comments are addressed

File: packages/cli/src/ui/components/InputPrompt.test.tsx (hunk 7 above)
Comment (Adib234) [reply packages/cli/src/ui/components/InputPrompt.test.tsx]: updated

File: packages/cli/src/ui/components/StatusDisplay.tsx (hunk 4 above)
Comment (Adib234) [reply packages/cli/src/ui/components/StatusDisplay.tsx:57]: Added check for history first

File: packages/cli/src/ui/components/InputPrompt.tsx (hunk 3 above)
Comment (Adib234) [reply packages/cli/src/ui/components/InputPrompt.tsx]: Added check in else case

Comment (jacob314) [reply packages/cli/src/ui/components/InputPrompt.tsx]: please revert both these checks for 
keyMatchers[Command.ESCAPE](key)
as line 453 is already checking that they key is escape so this just makes the logic confusing.

File: packages/cli/src/ui/components/StatusDisplay.tsx (hunk 4 above)
Comment (jacob314) [reply packages/cli/src/ui/components/StatusDisplay.tsx:57]: sorry I wasn't clear. I mean we should not show either if their is no history and the message is empty.

File: packages/cli/src/ui/components/InputPrompt.tsx (hunk 3 above)
Comment (Adib234) [reply packages/cli/src/ui/components/InputPrompt.tsx]: Removed check

File: packages/cli/src/ui/components/StatusDisplay.tsx (hunk 4 above)
Comment (Adib234) [reply packages/cli/src/ui/components/StatusDisplay.tsx:57]: updated check for not showing any toast when prompt is empty and there's no history

File: packages/cli/src/ui/components/InputPrompt.tsx [hunk 8]
```diff
@@ -508,10 +508,10 @@ export const InputPrompt: React.FC<InputPromptProps> = ({
         } else {
           // Second ESC
           resetEscapeState();
-          if (keyMatchers[Command.ESCAPE](key) && buffer.text.length > 0) {
+          if (buffer.text.length > 0) {
             buffer.setText('');
             resetCompletionState();
-          } else if (keyMatchers[Command.ESCAPE](key)) {
+          } else {
```
Comment (jacob314) [inline packages/cli/src/ui/components/InputPrompt.tsx:514]: nit: avoid double nesting. this can just be `else if (...`

**Review by jacob314 — APPROVED** [review]
lgtm

File: packages/cli/src/ui/components/InputPrompt.tsx (hunk 8 above)
Comment (Adib234) [reply packages/cli/src/ui/components/InputPrompt.tsx:514]: Will address in this PR https://github.com/google-gemini/gemini-cli/pull/15720
